libraries/ESP8266WiFi/src/CertStoreSPIFFSBearSSL.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+  CertStoreSPIFFSBearSSL.cpp - Library for Arduino ESP8266
+  Copyright (c) 2018 Earle F. Philhower, III
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include "CertStoreSPIFFSBearSSL.h"
+#include <FS.h>
+
+CertStoreSPIFFSBearSSL::CertStoreSPIFFSBearSSL() : CertStoreBearSSL()
+{
+}
+
+CertStoreSPIFFSBearSSL::~CertStoreSPIFFSBearSSL()
+{
+}
+
+CertStoreBearSSL::CertInfo CertStoreSPIFFSBearSSL::preprocessCert(const char *fname)
+{
+  CertStoreBearSSL::CertInfo ci;
+  memset(&ci, 0, sizeof(ci));
+
+  // Load the DER into RAM temporarially
+  File f = SPIFFS.open(fname, "r");
+  if (!f) return ci;
+  int sz = f.size();
+  uint8_t *buf = new uint8_t[sz];
+  if (!buf) {
+    f.close();
+    return ci;
+  }
+  f.read(buf, sz);
+  f.close();
+
+  ci = CertStoreBearSSL::preprocessCert(fname, buf, sz);
+
+  delete buf;
```

Comment:
Is this delete or delete[]?

Reply:
Should be delete[] for correctness, good catch.